support/scheduler/schedule.go

```diff
@@ -371,6 +375,9 @@ func triggerSchedule() {
 						loggingClient.Info("requeue schedule, detail : " + scheduleContext.GetInfo())
 						scheduleQueue.Add(scheduleContext)
 					}
+
+					//waiting the go routine finish
```

Comment:
It looks like this is all still effectively synchronous since we're blocking on the handling of one scheduleContext. The goroutine isn't really buying us much here.

Two thoughts come to mind:
1.) If you just call 
`go execute(scheduleContext, scheduleQueue)`
above, that should be sufficient. Since the scheduleContext is a reference, you can move the 
`scheduleContext.UpdateNextTime()`
`scheduleContext.UpdateIterations()`
into the execute function itself. And then passing the queue allows you to add the updated context.

This way if there are, say, 10 schedules you can fire off 10 goroutines concurrently. Although the way you determine whether your outer loop should continue will most likely need to be changed.
-OR -
2.) Pass the context into execute() and update the properties, sending the updated context out on a channel. The channel consumer is another goroutine adding contexts back to the queue as they come out.

For synchronizing operations around channels, take a look at sync.WaitGroup